format/nes/nes.jq

```diff
@@ -0,0 +1,9 @@
+def toasm:
+  select(.op_code and (.args // "")) | "\(.op_code) \(.args // "")";
+def tokitty(size):
+  def _sub: gsub("00"; "\u0022\u0022\u0022") | gsub("01"; "\u007F\u0022\u0022") | gsub("10"; "\u0022\u007F\u0022") | gsub("11"; "\u0022\u0022\u007F") | gsub(" "; "");
+  (select(.combined)
+  | "\(.combined | _sub)")
+  | "\\e_Ga=T,f=24,s=8,v=8,c=\(2*size),r=\(1*size);" + @base64 + "\\e\\\\\\n";
```

Comment:
I usually use "haskell"-ish jq style once jq program gets large, maybe something like:
```jq
def tokitty($size):
  def _sub:
    ( gsub("00"; "\u0022\u0022\u0022")
    | gsub("01"; "\u007F\u0022\u0022")
    | gsub("10"; "\u0022\u007F\u0022")
    | gsub("11"; "\u0022\u0022\u007F")
    | gsub(" "; "")
    );
  ( ( select(.combined)
    | "\(.combined | _sub)"
    )
  | "\\e_Ga=T,f=24,s=8,v=8,c=\(2*size),r=\(1*$size);" + @base64 + "\\e\\\\\\n"
  );
```

Is `.combined` always set for a `.chr_rom` struct? maybe can skip `select`?

A bit esoteric but size argument should probably be `$size` so that it is a value binding instead of a lambda. i.e. for `iskitty(1,2,3)` it should behave as `(1,2,3) as $a | iskitty($a)` not as `"...r=\1(1*(1,2,3));"` i guess?

Still debating what to about format specific addition functions. There is a bunch for mp4 and matroska that i've thought maybe move out some `mp4.jq`/`matroska.jq` library files and have in the wiki etc. And usage would then be something like `fq -L . 'include "mp4"; mp4_blabla' file.mp4`. Another option could be to look into jq modules and have namespace `mp4::blabla`, but i haven't used modules much yet.

Another concern is that currently to including of additinal jq code is not very performant, the AST will get cached but they will be compiled to byte code on each eval.

Reply:
I've prefixed them with `nes_` for now.

Reply:
👍 let embed them for now, maybe in the future we move them out or make it more efficient somehow